### PR TITLE
serve 200 OK for /v2/ API check

### PR DESCRIPTION
**Pull request by BenTheElder**
part of #77 

I think we could instead serve 401 and craft an appropriate `Www-Authenticate: Bearer realm` header that includes the scope (unlike GCR, which lets the client determine it based on the repo it parses in the image name), but that's more complex and unnecessary.

Our target use case involves publicly readable registries, and GCR public read is *really* public read, you don't need to present an auth token for any of the API calls.

### CODE REVIEW

File: cmd/archeio/app/handlers.go [hunk 1]
```diff
@@ -41,11 +41,17 @@ func MakeHandler(upstreamRegistry string) http.Handler {
 	blobs := newCachedBlobChecker()
 	doV2 := makeV2Handler(upstreamRegistry, blobs)
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// this registry only supports proxy image downloads / listing
+		// we do *not* want to deal with mutations, those are done against
+		// the backing registries by other tools
+		if r.Method != http.MethodGet && r.Method != http.MethodHead {
```
Comment (BenTheElder) [inline cmd/archeio/app/handlers.go:47]: unrelated tightening of the handler that I perhaps should have moved to another commit.

all of the endpoints we should support use one of these two methods https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints

POST/PUT/PATCH/DELETE are in the spec as well, but of course used for mutations (pushing images etc.), which we don't need or want to support here

Comment (BenTheElder) [reply cmd/archeio/app/handlers.go:47]: this needs test coverage, perhaps punt to another PR, it's not important.

**Comment by ameukam** [comment]
/lgtm
/hold

Remove the hold when ready.
